R/emissions_profile_any_compute_profile_ranking.R

```diff
@@ -61,10 +73,26 @@ flat_benchmarks <- function(data) {
 }
 
 add_rank <- function(data, .by) {
-  if (identical(.by, "all")) .by <- NULL
-  mutate(
-    data,
-    profile_ranking = rank_proportion(.data[[extract_name(data, aka("co2footprint"))]]),
-    .by = all_of(.by)
-  )
+  if (identical(.by, "all") | (identical(.by, "unit"))) {
+    if (identical(.by, "all")) .by <- NULL
+    mutate(
+      data,
+      profile_ranking = rank_proportion(.data[[extract_name(data, aka("co2footprint"))]]),
+      .by = all_of(.by)
+    )
+  } else {
+    if ((identical(.by, "isic_4digit")) | (identical(.by, c("unit", "isic_4digit")))) {
+      filtered_df <- data %>% filter(str_length(.data[["isic_4digit"]]) %in% c(NA_character_, 4, 5))
+      df <- anti_join(data, filtered_df, by = c("isic_4digit"))
+    } else {
+      filtered_df <- data %>% filter(is.na(.data[["tilt_sector"]]))
+      df <- anti_join(data, filtered_df, by = c("tilt_sector"))
+    }
+    output <- mutate(
+      df,
+      profile_ranking = rank_proportion(.data[[extract_name(data, aka("co2footprint"))]]),
+      .by = all_of(.by)
+    )
+    bind_rows(output, filtered_df)
+  }
```

Comment:
There could be a better logic to implement all the above usecases. I haven't much time to improve this logic as it could be the case that you will come up with a better one. Hence, I would like to leave this PR to you :)